frontend/components/subjects-component.js

```diff
@@ -9,41 +9,32 @@ class Subjects extends HTMLElement {
             </section>
         `;
 
-        const authToken = localStorage.getItem('authToken');
-        try {
-            const response = await fetch(`${Links.serverBaseURL}/subjects`, {
-                method: 'GET',
-                headers: {
-                    'Content-Type': 'application/json',
-                    'Authorization': `Bearer ${authToken}`
+        var _apiCall = new APICall();
+        
+        _apiCall.makeGetRequest('subjects')
+            .then(data=>{
+ 
+                if (data.data.length > 0) {
```

Comment:
May you please name the response data to `res` or `response` then update to `response.data.length` to avoid a chain of puzzlement